### ISSUE TITLE
Setting to filter/ignore drives within the same geofence (or according to certain user-set parameters)

### ISSUE THREAD

**Issue by fetzu** [adriankumpf/teslamate#618]
**Is your feature request related to a problem? Please describe.**

With the new Streaming API, every single and smallest drive is being recorded (which is great!) but it also records/counts the tiniest drives (for example coming in or out of a garage that has to be opened manually) as a drive entry (and destination). This means that for every coming in or out of garage (for example) will generate a single "0km" drive and essentially render some stats (median distance, top destinations...) null and void in the long run.

**Describe the solution you'd like**

It would be great to have a setting to "not record" (as in "not really count") drives which are within the same geofence (providing of course they meet other time. or distance-based criteria. This setting could be turned on on either the "settings" page or even within the geofence page; with maybe the time- or distance-based being set/customizable by the user.
It would either not count that data as a drive OR merge that data with the preceding or next drive (provided it started/ended within x time-frame).

**Describe alternatives you've considered**
Might be possible to filter out this data in Grafana in some way (as in "not show up drives less than x km" for example), but I think the data would still pollute other stats (top destinations, median drive length...)

**Additional Context**
Example of these "0 km" drives.
<img width="733" alt="NullDrives" src="https://user-images.githubusercontent.com/6372605/79792420-2551e300-834f-11ea-9ba2-d6b7dc26e5fd.PNG">

**Comment by jmiverson** [comment]
After seeing this, I checked my Drives and don't see any like those you're showing.  I'd have expected a few (I've been running 1.19-dev and now latest for a couple weeks).  Wonder what the difference is.  I moved my car yesterday from one side of the driveway to the other.  No drive was recorded for that.  Anything unique about your short drives?

**Comment by DariBer** [comment]
I get the same problem, it loggs small trips in my garage.

<img width="1435" alt="Screenshot 2020-04-20 at 22 05 34" src="https://user-images.githubusercontent.com/26138185/79795038-9bf0df80-8353-11ea-8061-fe5b72f1bf0e.png">
<img width="1228" alt="Screenshot 2020-04-20 at 22 05 48" src="https://user-images.githubusercontent.com/26138185/79795048-9f846680-8353-11ea-8e45-e91b7fa46136.png">

**Comment by krezac** [comment]
I'd say:
1. if less then 5 minutes from end of last trip, append to the last trip
2. if shorter than 500m (and not above), ignore
3. start new trip

**Comment by gilrim** [comment]
These are sensible defaults, but should preferably be configurable from settings page.

**Comment by fetzu** [comment]
> 
> 
> Anything unique about your short drives?

I don't think so, these are really short 4-5 meters trip out of the garage. If it can help narrow it down I am on a S with MCU1 and also using TeslaFi.

**Comment by Dulanic** [comment]
> These are sensible defaults, but should preferably be configurable from settings page.

I agree, and also I would say the setting should purely be on the report side not on the data storage side. But if that's the case, maybe just editing the chart personally within grafana is a viable option instead?

**Comment by adriankumpf** [comment]
I absolutely agree that this should be implemented purely be on the report side. This requires only minimal adjustments to the queries.

**Comment by Dulanic** [comment]
@fetzu BTW you can simply update your grafana and just add to set a min of 1 km or change to however you feel best. He wasn't exaggerating when @adriankumpf  said minimal :) All you need to do is go to grafana, click edit from the drop down of the panel to edit /w admin login.

`WHERE distance > 1;`

![image](https://user-images.githubusercontent.com/2948206/80099928-dcf41a00-8534-11ea-86d1-ac6596b76e7a.png)

**Comment by MaxG88** [comment]
> I'd say:
> 
> **1. if less then 5 minutes from end of last trip, append to the last trip**
> 2. if shorter than 500m (and not above), ignore
> 3. start new trip

Second point should be doable on Grafana, but what about the first one? I think it's a valid point and as was said before, being able to configure it in the Settings page would be great.

**Comment by fetzu** [comment]
@Dulanic Thank you for the quick tip, but this does not solve the scope of the problem at all: changing how drives are shown on the dashboard will not refrain these "end destinations" to be recorded and shown in other dashboards such as "Locations" or "Drive stats". 

I understand I could manually change the settings of each of these dashboards in Grafana to show the data the way I want to; but this is not in any way a permanent solution (ie: that modified dashboard might become obsolete with every release).

I am in no way proficient nor comfortable enough to code and implement this, so I don't know what exactly could be done in Grafana or not (user-set settings for example), or if this need to be taken care of upstream.

**Comment by Dulanic** [comment]
So I don't have the time to do much with this.,.. but here is a function that would adjust the drives to a custom rounded minute range,..... 

[rounded_drive.zip](https://github.com/adriankumpf/teslamate/files/4529508/rounded_drive.zip)

@MaxG88 Here is a quick function I made and an example of how to use it.,... this would start times to the nearest 30 minutes. I set the where to show specific examples where it would combine the drives. So I don't think we need to change the data storage method as we can do it on the report side. It's also changeable via the function variable.

![image](https://user-images.githubusercontent.com/2948206/80224410-447e9800-860f-11ea-97ed-db704a26a586.png)
